tests/unit/test_types.py

```diff
@@ -297,11 +305,14 @@ def test_strip_unnecessary_quotes():
 
 
 def test_python_type_to_snow_type():
-    def check_type(python_type, snow_type, is_nullable):
+    # In python 3.10, the __name__ of nested type only contains the parent type, which breaks our test. And for this
+    # reason, we introduced type_str_override to test the expected string.
+    def check_type(python_type, snow_type, is_nullable, type_str_override=None):
         assert python_type_to_snow_type(python_type) == (snow_type, is_nullable)
-        assert python_type_to_snow_type(
-            getattr(python_type, "__name__", str(python_type))
-        ) == (snow_type, is_nullable)
+        type_str = type_str_override or getattr(
```

Comment:
What if we just use `str(python_type)` instead of `__name__`? Python 3.10.9 on my machine evaluates:
`str(typing.Union[str, None])` to `'typing.Optional[str]'`, which is what we want right?

Reply:
Ah I'm not aware of this, let me try `str(python_type)` and see how it works :)

Reply:
Hmmm it looks like it does not work perfectly for basic types such as `int`, which gives `<class 'int'>` 🤦

Reply:
Looks like there is `typing.Type`. Let me try that.

Reply:
`type(typing.Optional[str])` gives `<class 'typing._GenericAlias'>`. I think this is beyond our grasp to be honest 🤔

Reply:
I see... I just don't like supplying literal strings and not explaining how to obtain them. Especially, now there're at least three cases (1) type (2) TypeVar (3) other type hints. I don't have a strong opinion otherwise.

Reply:
To be honest I think we could make them consistence by always providing a string :p